### PR TITLE
[sdk] update to Gecko SDK 4.3.1

### PR DESCRIPTION
**Pull request by lmnotran**

### CODE REVIEW

**Review by suveshpratapa — APPROVED** [review]
LGTM

File: examples/sleepy-demo/sleepy-mtd.c [hunk 1]
```diff
@@ -58,8 +58,8 @@
 #define MULTICAST_PORT 123
 #define RECV_PORT 234
 #define SLEEPY_POLL_PERIOD_MS 2000
-#define MTD_MESSAGE "mtd button"
 #define FTD_MESSAGE "ftd button"
+#define MTD_MESSAGE "mtd button"
```
Comment (parag-silabs) [inline examples/sleepy-demo/sleepy-mtd.c:62]: This might cause conflicts since another PR also have the same/similar changes.

Comment (lmnotran) [reply examples/sleepy-demo/sleepy-mtd.c:62]: This is an easy conflict to resolve. What PR are you referring to?

Comment (lmnotran) [reply examples/sleepy-demo/sleepy-mtd.c:62]: Oh nevermind, you're referring to my sleepy-demo-ssed PR. It'll be fine. I can take care of the conflict if it comes up

**Review by parag-silabs — APPROVED** [review]
LGTM.

**Review by romacdon — APPROVED** [review]
LGTM

**Review by jwhui — APPROVED** [review]

**Pull request merged**